### PR TITLE
fixed problems with add, list, rank commands

### PR DESCRIPTION
**Pull request by jung1411**
# Befor putting PR (put x in bracket if you ran it.);

- [x] Run `pylint $(git ls-files '*.py')` passes

# Overview
Issue https://github.com/dhmoon91/discord/issues/33

Fixed problems with commands: `add`, `list`, `display` commands:

1. Problem with add command - no rank number value (I, II, III, IV)
- League Points (LP) was added in order to make team for top tiers like Master, Grand Master and Challengers as league points were necessary to differentiate the gap between summoners with same tier

2. Problem with list command: Challenger, Grand Master and Master players shown as C1, G1, M1 as shown below

   image
- Changed it to `Challenger: CH`, `Grand Master: GM`, `Master: MA` so that it is easier to recognize

   image

3. Problem with rank command: `@관전남 <command>` not needed and changes needed in positioning as shown below

   image
- Removed the `@관전남 <command>` and changed position between the rank information and `All Data from NA server`

   image
   
   image

4. Problem with rank command: Had errors when trying to display rank for Unranked summoners as shown below

   image
- Added emblem for unranked tier, set all rank values to zero if unranked as it was complaining how they can't find rank information

   image

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -116,9 +121,8 @@ async def get_rank(ctx, *, name: str):  # using * for get a summoner name with s
 
         embed_data = EmbedData()
         embed_data.title = "Solo/Duo Rank"
-        embed_data.description = (
-            f"`All Data from NA server`\n\n <@!{bot.user.id}> <command>"
-        )
+
+        # embed_data.description = f"<@!{bot.user.id}> <command>"
```
Comment (dhmoon91) [inline bot.py]: delete this

File: bot.py [hunk 2]
```diff
@@ -141,22 +145,31 @@ async def get_rank(ctx, *, name: str):  # using * for get a summoner name with s
...
         summoner_total_game = summoner_info["solo_win"] + summoner_info["solo_loss"]
-        solo_rank_win_percentage = int(
-            summoner_info["solo_win"] / summoner_total_game * 100
+
+        # Due to zero division error, need to handle situation where total games are zero
+        if summoner_total_game == 0:
+            solo_rank_win_percentage = 0
+        else:
+            solo_rank_win_percentage = int(
+                summoner_info["solo_win"] / summoner_total_game * 100
```
Comment (dhmoon91) [inline bot.py]: I think you can handle this one line - unless you need if/else block for other reason
`solo_rank_win_percentage =  0 if summoner_total_game == 0 else  int(summoner_info["solo_win"] /summoner_total_game * 100`

'true' if True else 'false'

**Review by dhmoon91 — APPROVED** [review]
no issue from me but think of making code simpler if you can before merging

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
 node_modules
 .vscode/
 .env
+.idea
```
Comment (smoondh) [inline .gitignore:4]: ?

Comment (jung1411) [reply .gitignore:4]: 뭐 extension 다운받았는데 저거 생김

File: bot.py [hunk 4]
```diff
@@ -359,11 +370,19 @@ async def display_current_list_of_summoners(ctx):
...
+                        file_data[server_id][count]["tier_division"]
+                    ),
+                    file_data[server_id][count]["formatted_user_name"],
+                )
+            else:
+                output_str += "`{0}{1}` {2}\n".format(
+                    file_data[server_id][count]["tier_division"][0],
+                    TIER_RANK_MAP.get(file_data[server_id][count]["tier_rank_number"]),
+                    file_data[server_id][count]["formatted_user_name"],
+                )
```
Comment (smoondh) [inline bot.py:385]: use ternary operator here too

**Review by jjaykim — APPROVED** [review]

**Pull request merged**